README.md

```diff
@@ -1,2 +1,105 @@
 # sampler
 public infrastructure tools
+
+
+## Overview
+
+This repository provides a Bash script 
+(`get-aws-subaccount-session.sh`) to establish 
+an AWS CLI session with a subaccount using 
+role assumption. It allows users to securely 
+switch AWS accounts using MFA authentication, 
+ensuring temporary access without storing 
+permanent credentials.
+
+The script:
+* Assumes an IAM role in a subaccount using 
+AWS STS.
+* Requires MFA authentication to generate 
+temporary session tokens.
+* Supports manual or automated MFA entry 
+(if a TOTP secret is stored).
+* Ensures AWS credentials persist in the 
+terminal session.
+
+### Prerequisites
+
+Before using this script, ensure that:
+* You have AWS CLI installed and configured 
+with your base account credentials.
```

Comment:
Include a direct link to https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html, because there are old instructions out there that will get an old version, like `pip install awscli` which still works but installs v1.